File: backend/backend/Domain/Entities/Trip.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace backend.Domain.Entities;

public class Trip
{
    [BsonId]
    [BsonElement("_id"),BsonRepresentation(BsonType.ObjectId)]
    public string? Id {get; set;}
    
    [BsonElement("title")] 
    public string? Title;
    
    [BsonElement("created_by"),BsonRepresentation(BsonType.ObjectId)]
    public string? CreatedBy {get; set;}
    
    [BsonElement("created_on")]
    public DateTime CreatedOn {get; set;}
    
    [BsonElement("destinations")]
    public List<Destination> Destinations {get; set;}
    
    [BsonElement("links")]
    public List<Link> Links {get; set;}
    
    [BsonElement("days")]
    public List<Day> Days {get; set;}
    
    [BsonElement("expenses")]
    public List<Expense> Expenses {get; set;}

}

File: frontend/src/models/Trip.ts
import { Day } from "./Day"
import { Destination } from "./Destination"
import { Expense } from "./Expense"
import { Link } from "./Link"

export type Trip = {
    Id:             string,
    Title?:         string,
    CreatedBy:      string,
    CreatedOn:      Date,
    Destinations:   Destination[]
    Links:          Link[],
    Days:           Day[],
    Expenses:       Expense[]
}
